### PR TITLE
Fix incorrect logic when tokenising email headers

### LINKED ISSUE #12871: Unable to send e-mails, Uncaught PHP Exception TypeError: "str_ireplace()
### Mautic Version

5.0.x series

### PHP version

8.0.30

### What browsers are you seeing the problem on?

Not relevant

### What happened?

500 Internal Server Error when trying to send e-mails

### How can we reproduce this issue?

Step 1: Try to send e-mail via Channels - EMails - Send Example
Step 2: Campaign e-mails are affected as well

### Relevant log output

```shell
=> /var/www/mautic/var/logs/mautic_prod-2023-11-03.php <==
[2023-11-03 20:16:49] mautic.CRITICAL: Uncaught PHP Exception TypeError: "str_ireplace(): Argument #3 ($subject) must be of type array|string, Symfony\Component\Mime\Address given" at /var/www/mautic/app/bundles/EmailBundle/Helper/MailHelper.php line 699 {"exception":"[object] (TypeError(code: 0): str_ireplace(): Argument #3 ($subject) must be of type array|string, Symfony\\Component\\Mime\\Address given at /var/www/mautic/app/bundles/EmailBundle/Helper/MailHelper.php:699)"} {"hostname":"xxx","pid":6739}
```

`[2023-11-04 13:19:02] console.CRITICAL: Error thrown while running command "mautic:campaigns:trigger". Message: "str_ireplace(): Argument #3 ($subject) must be of type array|string, Symfony\Component\Mime\Address given" {"exception":"[object] (TypeError(code: 0): str_ireplace(): Argument #3 ($subject) must be of type array|string, Symfony\\Component\\Mime\\Address given at /var/www/mautic/app/bundles/EmailBundle/Helper/MailHelper.php:699)","command":"mautic:campaigns:trigger","message":"str_ireplace(): Argument #3 ($subject) must be of type array|string, Symfony\\Component\\Mime\\Address given"} {"hostname":"xxx","pid":2444}`

### Code of Conduct

- [X] I confirm that I have read and agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by mollux**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [x]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #12871 

#### Description:

see #12871

when setting a custom `return-path` header, an error 500 occurs.

This is because the logic in `MailHelper::searchReplaceTokens()` is not working correct anymore since the switch to Symfony mailer. The headers are no longer (arrays of) plain text values, but (lists of) various objects (e.g. an `Address` object) and types. For some headers like `return-path` it even throws an error.

This PR changes the token support for mail headers to only process `UnstructuredHeader` instances.

#### Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. set a custom `return-path` (see the referenced issue for details)
3. Send en email
4. validate it's successfully sent.

### CODE REVIEW

File: app/config/local.php.backup [hunk 1]
```diff
@@ -0,0 +1,786 @@
+<?php
```
Comment (escopecz) [inline app/config/local.php.backup]: @mollux was this file merged by accident?

Comment (mollux) [reply app/config/local.php.backup]: indeed, fixed

**Review by escopecz — APPROVED** [review]
I can't imagine how the headerBody could ever be an array either.  Too bad there is no test for it. Approving code-review 👍

**Review by patrykgruszka — APPROVED** [review]
This fixes issue #12871. Replacing tags in custom headers also works correctly after this change. The code looks good.

**Comment by Tie-fighter** [comment]
Thank you for fixing #12871 !

I have a silly question though...
What is the best way to update my installation to receive this fix?

**Pull request merged**